libraries/AP_Baro/AP_Baro_MS5611.cpp

```diff
@@ -516,5 +567,32 @@ void AP_Baro_MS56XX::_calculate_5837()
 
     _copy_to_frontend(_instance, (float)pressure, temperature);
 }
+// Calculate Temperature and compensated Pressure in real units (Celsius degrees*100, mbar*100).
+void AP_Baro_MS56XX::_calculate_5837_02ba() {
+    int32_t dT = _D2 - ((int32_t)_cal_reg.c5 << 8);
+    int32_t TEMP = 2000 + ((dT * _cal_reg.c6) >> 23);
+
+    int64_t OFF = ((int64_t)_cal_reg.c2 << 17) + (((int64_t)_cal_reg.c4 * dT) >> 6);
+    int64_t SENS = ((int64_t)_cal_reg.c1 << 16) + (((int64_t)_cal_reg.c3 * dT) >> 7);
+
+    if (TEMP < 2000) {
+        // Second-order compensation
+        int32_t T2 = (dT * dT) >> 31;
+        int64_t TEMP_low = TEMP - 2000;
+        int64_t OFF2 = 5 * (TEMP_low * TEMP_low) >> 1;
+        int64_t SENS2 = (TEMP_low * TEMP_low) >> 3;
+
+        TEMP -= T2;
+        OFF -= OFF2;
+        SENS -= SENS2;
+    }
+
+    // Cast _D1 to int64_t before performing multiplication and shift
+    int64_t pressure = ((((int64_t)_D1 * SENS) >> 21) - OFF) >> 15;
+
+    // Update frontend with calculated values
+    _copy_to_frontend(_instance, (float)pressure / 10, (float)TEMP / 100);
```

Comment:
```suggestion
    _copy_to_frontend(_instance, (float)pressure, (float)TEMP / 100);
```
`pressure` is already in `mbar*100` at this point.